bin/dump_data_to_file.py

```diff
@@ -33,30 +34,26 @@ def retrieve_data_from_server(datastore_url, user, key, start_ts, end_ts):
     print(f"Found {len(ret_list)} entries")
 
 
-def main():
-    # parse command line arguments
+def parse_args():
     parser = argparse.ArgumentParser()
 
-    parser.add_argument("--user", type=str, required=True)
-    parser.add_argument("--endpoint", type=str, required=True)
-    
     parser.add_argument("--datastore-url", type=str, default="http://localhost:8080")
-    parser.add_argument("--start-ts", type=float, default=0)
-    parser.add_argument("--end-ts", type=float, default=arrow.get().timestamp)
+    parser.add_argument("--spec-id", type=str)
+
+    # if one of these arguments is specified, the others in this group must also be specified
+    if any(arg in sys.argv for arg in ["--key", "--user", "--start-ts", "--end-ts"]):
+        parser.add_argument("--key", type=str, required=True)
+        parser.add_argument("--user", type=str, required=True)
+        parser.add_argument("--start-ts", type=float, required=True)
+        parser.add_argument("--end-ts", type=float, required=True)
```

Comment:
Is this the recommended option for conditional arguments? I know that you can define mutually exclusive groups so users can't specify `--user` and `--key` at the same time.